MangaManager/src/MetadataManager/CoverManager/CoverManager.py

```diff
@@ -267,15 +315,63 @@ def serve_gui(self):
             comic_frame.grid()
         self.redraw(None)
 
-    def select_frame(self, _, frame: ComicFrame, pos):
-        print(pos)
+    def _compare_images(self, selected_image, x, compared_image, comicframe, pos):
+        delta = float(self.delta_entry.get())
+        y = np.array(compared_image.histogram())
+        if compare_image(selected_image, compared_image, x, y, delta=delta):
+            self.select_frame(None, frame=comicframe, pos=pos)
+
+    def select_similar(self):
+        """
+        Compares the selected file with all the loaded covers and backcovers
+        Selects files that match.
+        :return:
+        """
+        assert len(self.selected_frames) == 1
+        frame, pos = self.selected_frames[0]
+        if pos == "front":
+            selected_photoimage: ImageTk.PhotoImage = frame.loaded_cinfo.get_cover_cache()
+        else:
+            selected_photoimage: ImageTk.PhotoImage = frame.loaded_cinfo.get_backcover_cache()
+
+        selected_image = ImageTk.getimage(selected_photoimage)
+        x = np.array(selected_image.histogram())
+        self.clear_selection()
+        # Compare all covers:
+        for comicframe in self.scrolled_widget.winfo_children():
+            try:
+                comicframe: ComicFrame
+                lcinfo: LoadedComicInfo = comicframe.loaded_cinfo
+                if self.scan_covers.get():
```

Comment:
Front and back are the same code except some accessors and strings, you can refactor to a private method:

def compare_image(self, enabled, image, "front"):
   pass
   
compare_image(self.scan_backcovers.get():, image, "back")